src/components/chart components/LineChart.jsx

```diff
@@ -0,0 +1,56 @@
+import React, { useState, useEffect } from 'react';
+import { generateChartStyle, generateOptions } from "./chartProperties";
+import { generateChartData, dateFormatterShort, generateLabels } from "./chartsFunctions";
+
+import {
+    Chart as ChartJS,
+    CategoryScale,
+    LinearScale,
+    PointElement,
+    LineElement,
+    Title,
+    Tooltip,
+} from 'chart.js';
+import { Line } from 'react-chartjs-2';
+
+ChartJS.register(
+    CategoryScale,
+    LinearScale,
+    PointElement,
+    LineElement,
+    Title,
+    Tooltip,
+);
+
+
+export default function LineChart({ timeInterval, chartTitle, startDate, endDate }) {
+    const [chart, setChart] = useState([]);
+    const url = "http://localhost:3004/transactions";
+    useEffect(() => {
+        fetchTransactions();
+    }, []);
+
+    const fetchTransactions = async () => {
+        try {
```

Comment:
same comment as above, this should use the generic transactions api. and should be in fact two components, a container and a presentational one